File: Problems & Solutions/G/Geeks And The String/23 January 2023/Geeks And The String.cpp
//{ Driver Code Starts
#include<bits/stdc++.h>
using namespace std;


// } Driver Code Ends
class Solution {
  public:
    string removePair(string s) {
        
        int j = -1;
        
        for(int i=0; i<s.size(); i++)
        {
            if(j >= 0 && s[j] == s[i]) j--;
            else s[++j] = s[i];
        }
        
        return j < 0 ? "-1" : s.substr(0, j+1);
    }
};


//{ Driver Code Starts.

int main(){
    int t;
    cin>>t;
    while(t--){
        string s; 
        cin>>s;
        
        Solution obj;
        string res = obj.removePair(s);
        
        cout<<res<<endl;
        
    }
}

// } Driver Code Ends

File: Problems & Solutions/G/Geeks And The String/23 January 2023/Geeks And The String.java
//{ Driver Code Starts
import java.io.*;
import java.util.*;

class GFG {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int t;
        t = Integer.parseInt(br.readLine());
        while(t-- > 0){
            
            String s; 
            s = br.readLine();
            
            Solution obj = new Solution();
            String res = obj.removePair(s);
            
            System.out.println(res);
            
        }
    }
}

// } Driver Code Ends


class Solution {
    public static String removePair(String s) {
        
        Stack<Character>p= new Stack<>();
        char c;
        String ans = "";
        int n = s.length();

        for(int i=0; i<n; i++)
        {
           if(p.empty())
           {
               p.push(s.charAt(i));
           }

           else
           {
               if(p.peek() == s.charAt(i))
               {
                   p.pop();
               }
               else p.push(s.charAt(i));
           }
        }

        while(!p.empty())
        {
            c = p.peek();
            p.pop();
            ans += c;
        } 
        
        StringBuilder str = new StringBuilder(ans);
        StringBuilder reverseStr = str.reverse();
        String result = str.toString();

        if(ans.length() == 0)
        {
            return "-1";
        }

        return result;
    }
}
